src/mmw/js/src/modeling/views.js

```diff
@@ -326,7 +326,14 @@ var ScenarioTabPanelView = Marionette.ItemView.extend({
     },
 
     renameScenario: function() {
-        var self = this;
+        var self = this,
+            setScenarioName = function(name) {
+                if (name && name !== '' && name !== self.model.get('name')) {
```

Comment:
Also, we [check if the name is the same inside `updateScenarioName`](https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/js/src/modeling/models.js#L511) so checking it here in the `if` condition is redundant. Removing that too ...